src/main/java/org/terasology/biomesAPI/BiomeManager.java

```diff
@@ -128,11 +152,26 @@ public void checkBiomeChangeEvent(MovedEvent event, EntityRef entity) {
             if (!oldBiomeOptional.isPresent()) {
                 return;
             }
+
             Biome newBiome = newBiomeOptional.get();
             Biome oldBiome = oldBiomeOptional.get();
             if (oldBiome != newBiome) {
                 entity.send(new OnBiomeChangedEvent(oldBiome, newBiome));
+                metricsMode.setBiome(newBiome.getId());
             }
         }
     }
+
+    @ReceiveEvent(components = PlayerCharacterComponent.class)
+    public void checkPlayerSpawnedEvent(OnPlayerSpawnedEvent event, EntityRef entity) {
+        LocationComponent locationComponent = entity.getComponent(LocationComponent.class);
+        Vector3i spawnPos = new Vector3i(locationComponent.getWorldPosition());
+        final Optional<Biome> spawnBiomeOptional = getBiome(spawnPos);
+        if (!spawnBiomeOptional.isPresent()) {
+            return;
+        }
+        Biome spawnBiome = spawnBiomeOptional.get();
+        metricsMode.setBiome(spawnBiome.getId());
```

Comment:
I think this can be done in more idiomatic (functional) code using the `Optional`:

```suggestion
        getBiome(spawnPos)
            .ifPresent(spawnBiome -> metricsMode.setBiome(spawnBiome.getId()));
```

Reply:
As explanation: First, we can rewrite the code block to not use an early `return` statement:

```java
        final Optional<Biome> spawnBiomeOptional = getBiome(spawnPos);
        if (spawnBiomeOptional.isPresent()) {
	        Biome spawnBiome = spawnBiomeOptional.get();
	        metricsMode.setBiome(spawnBiome.getId());
        }
```

With this, we're using the `Optional` as we would do normally by conducting a `null` check (here, `ifPresent()`)  and afterwards using the value (here, `.get()` it first). 

Semanticall, we express that, if the value is present, we want to do some _computation_ - that is what can be done using `ifPresent`. We are to define a lambda expression for the _computation_ we want to perform. Basically, the `spawnBiome` is automatically "unwrapped" fro m the Optional:

```java
        final Optional<Biome> spawnBiomeOptional = getBiome(spawnPos);
        spawnBiomeOptional.ifPresen(spawnBiome -> {
        	metricsMode.setBiome(spawnBiome.getId()
		});
```

This can be further inlined to end up with above suggestion.